### PR TITLE
Make new native queries default to using the last-selected database

### PR DESCRIPTION
**Pull request by kdoh**
Implements #8780 by setting a localStorage item with the last selected database id if a user is creating native queries.

### CODE REVIEW

File: frontend/src/metabase/query_builder/actions.js [hunk 1]
```diff
@@ -281,6 +281,16 @@ export const initializeQB = (location, params) => {
           card = null;
         }
 
+        // if we're starting a native query and  a recent DB has been set grab it and then use it to start a new
+        // query based off of that DB
+        const recentDb = window.localStorage.getItem(
+          "METABASE_LAST_USED_DB_ID",
```
Comment (tlrobinson) [inline frontend/src/metabase/query_builder/actions.js]: Maybe make this a `const`

**Review by tlrobinson — APPROVED** [review]
LGTM but if you feel like making that key a const that might be nice.

**Comment by mazameli** [comment]
@tlrobinson Don't know if you tried this out in the branch, but it has some problems when selecting saved SQL-based questions, i.e. it's either showing the New Question screen, or it'll show an empty editor:

existing-qs

**Comment by tlrobinson** [comment]
@mazameli You're right, retracting my LGTM.

File: frontend/src/metabase/query_builder/actions.js [hunk 2]
```diff
@@ -281,6 +281,16 @@ export const initializeQB = (location, params) => {
...
+        );
+
+        if (card.dataset_query.type === "native" && recentDb) {
```
Comment (tlrobinson) [inline frontend/src/metabase/query_builder/actions.js]: Looks like this blows away existing cards.

**Review by tlrobinson — CHANGES_REQUESTED** [review]
We need to inject the recent db ID in a different place. Currently it blows away saved/serialized cards.

I think the two ways to get to a new native query are the new question flow, and clicking "Switch to SQL" before selecting a database (though that currently defaults to the first DB for some reason)

**Comment by mazameli** [comment]
Would be cool to figure out how to keep this from blowing away existing questions. It'd be a nice little UX lift.

**Comment by kdoh** [comment]
Declaring PR bankruptcy on this one.

**Comment by mazameli** [comment]
image

**Pull request closed without merging**